### DISCUSSION TITLE
Full navigation of LaTeX code with smartparens

### DISCUSSION THREAD

**Discussion by DrWaleedAYousef** [Fuco1/smartparens#1191] [Q&A]
## Expected behavior
How can I configure `smartparens` to treat `\begin ...\end` environments, and commands, e.g., `\textbf`, delimiters so that I can navigate them, e.g., using `sp-forward-sexp`.

## Environment & version information

**In recent enough `smartparens` you can call `M-x sp-describe-system` to generate this report**.  Please fill manually what we could not detect automatically.  Edit the output as you see fit to protect your privacy.

- Active major-mode: LaTeX

- Emacs version (`M-x emacs-version`): GNU Emacs 26.1 (build 1, x86_64-pc-linux-gnu, GTK+ Version 3.22.30) of 2018-07-05

- Spacemacs/Evil/Other starterkit (specify which)/Vanilla: The three appear when pressing sp-describe-system

- OS: ArchLinux last update (as today)

**Comment by Fuco1** [comment]
since the begin/end environments depend on the argument it is currently not possible to configure (we only have static pairs).  I don't know what you mean by `\textbf` but you can look in `smartparens-latex.el` to see how pairs are set up.

**Comment by DrWaleedAYousef** [comment]
Ok, thanks so much. Regarding my second part of the question (probably it is not related to LaTeX commands; I am not sure). This is an example; suppose you have the next line:

This is a brackets (word). This is the first line. Another word `\textbf{Bold Text}`|

Now, the cursor is at | and I hit C-M-b, cursor comes at the first `{`, which is good; but another C-M-b brings the cursor to the first `(`.  However, if the cursor is after "another":

This is a brackets (word). This is the first line. Another| word `\textbf{Bold Text}`

hitting C-M-b will treat each word as a regexp and navigate back through words. So, why it does not treat words consistently as regexp and why it does not treat the LaTeX command `\textbf` as a regexp as well.
